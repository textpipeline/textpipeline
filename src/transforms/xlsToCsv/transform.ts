import { Transform } from '../types';

const xlsToCsv: Transform = {
  name: 'XLS to CSV',
  slug: 'xls-to-csv',
  inputType: 'XLS',
  outputType: 'CSV',
  project: {
    name: 'xlsx',
    projectHref: 'https://sheetjs.com/opensource',
    repositoryHref: 'https://github.com/SheetJS/js-xlsx',
    license: 'apache-2.0',
  },
  execute: input => Promise.resolve(`TEST OUTPUT: ${input}`),
};

export default xlsToCsv;
