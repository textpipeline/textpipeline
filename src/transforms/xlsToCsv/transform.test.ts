import xlsToCsv from './transform';

describe('XLS to CSV', () => {
  it('has the correct attributes', () => {
    expect(xlsToCsv).toMatchSnapshot();
  });

  describe('when called with valid input', () => {
    let output: string;

    beforeEach(async () => {
      output = await xlsToCsv.execute('stuff');
    });

    it('produces the correct output', () => {
      expect(output).toMatchSnapshot();
    });
  });
});
