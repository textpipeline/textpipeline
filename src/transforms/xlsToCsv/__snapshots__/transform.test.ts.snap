// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`XLS to CSV has the correct attributes 1`] = `
Object {
  "execute": [Function],
  "inputType": "XLS",
  "name": "XLS to CSV",
  "outputType": "CSV",
  "project": Object {
    "license": "apache-2.0",
    "name": "xlsx",
    "projectHref": "https://sheetjs.com/opensource",
    "repositoryHref": "https://github.com/SheetJS/js-xlsx",
  },
  "slug": "xls-to-csv",
}
`;

exports[`XLS to CSV when called with valid input produces the correct output 1`] = `"TEST OUTPUT: stuff"`;
